### ISSUE TITLE
[RF] Move methods to different class

### ISSUE THREAD

**Issue by fabio-oesch** [JCKing97/Agents4Asteroids#1]
I think there are a few methods in `game_control.py` which could be moved to `game_entitities.py`.

The methods in question are:
```python3
def particle_update(self, window, particles)
def asteroid_generate(self, window)
def entity_update(self, window_width, window_height, particles, asteroids, ship)
```

We already have certain update methods in the `game_control.py`.

**Comment by JCKing97** [comment]
I agree there was a bit of refactoring to do in terms of the particle_update method, but I'm happy for now where asteroid_generate is and entity_update

**Comment by JCKing97** [comment]
Some refactoring done here: https://github.com/autonomy-lab/Agents4Asteroids/pull/5

**Comment by JCKing97** [comment]
This still needs more refactoring...
